src/sidebar/services/test/frame-sync-test.js

```diff
@@ -591,6 +604,30 @@ describe('FrameSyncService', () => {
       await connectGuest();
       assert.calledWith(channel.call, 'setHighlightsVisible', false);
     });
+
+    [true, false].forEach(haveContentInfo => {
+      it('sends content info to guest if available', async () => {
+        let channel;
+        setupPortRPC = rpc => {
+          channel = rpc;
+        };
+        const contentInfo = { item: { title: 'Some article' } };
+        if (haveContentInfo) {
+          fakeStore.setContentInfo(contentInfo);
+        }
+
+        await connectGuest();
+        emitGuestEvent('documentInfoChanged', {
+          uri: 'https://publisher.org/article.pdf',
+          frameIdentifier: null,
+        });
```

Comment:
Can you help me understand the relevance of this?

Based on https://github.com/hypothesis/client/blob/1eee7a77765ac459e11fa0beebcc4011551e255e/src/sidebar/services/frame-sync.js#L279-L294 it looks like maybe this has to happen for the frame to actually connect up and get an identifier? A little tough to follow on the testing side of things!

Perhaps at some point these tests could be refactored to be clearer (or `frame-sync` itself refactored...)?

Reply:
> it looks like maybe this has to happen for the frame to actually connect up and get an identifier? A little tough to follow on the testing side of things!

Good question! The `documentInfoChanged` event is actually obsolete here. It was needed in an earlier iteration when `showContentInfo` was only sent to certain guests (the one with `frameIdentifier` set to `null`). That is no longer the case, so I removed these lines.